### PR TITLE
MGMT-14409: generate short image URL

### PR DESCRIPTION
**Pull request by adriengentil**
Generate short URL in the infraenv to download discovery ISO, see the [enhancement](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/short-url-enhancement.md) for more details.

Test locally if there are no issues updating an infraenv from the former URL to the short URL. Tested also the opposite if we do a rollback.

### CODE REVIEW

File: internal/imageservice/url_test.go [hunk 1]
```diff
@@ -115,6 +126,43 @@ var _ = Describe("URL building", func() {
...
+		Expect(err).NotTo(HaveOccurred())
+
+		parsedISOType, parsedArch, parsedVersion, err := ParseDownloadURL(u)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(parsedISOType).To(Equal(isoType))
+		Expect(parsedArch).To(Equal(arch))
+		Expect(parsedVersion).To(Equal(version))
+	})
+
+	It("successfully parse malformed URL", func() {
```
Comment (osherdp) [inline internal/imageservice/url_test.go:160]: it should successfully parse a malformed URL?

File: internal/bminventory/inventory_v2_handlers.go [hunk 2]
```diff
@@ -778,20 +778,51 @@ func (b *bareMetalInventory) GetInfraEnvDownloadURL(ctx context.Context, params
...
 		return "", nil, err
 	}
-	return urlString, expiresAt, nil
+
+	shortURL, err := imageservice.ShortImageURL(b.ImageServiceBaseURL, imageservice.ByTokenPath, token, version, arch, imageType)
+	return shortURL, exp, err
+}
+
+func (b *bareMetalInventory) generateShortImageDownloadURLByID(infraEnvID, imageType, version, arch string) (string, *strfmt.DateTime, error) {
+	var expiresAt strfmt.DateTime
```
Comment (osherdp) [inline internal/bminventory/inventory_v2_handlers.go:822]: we don't need to set expiration date to some value?

File: internal/imageservice/url_test.go (hunk 1 above)
Comment (adriengentil) [reply internal/imageservice/url_test.go:160]: maybe I should call it `successfully parse a non-infraenv URL`? as the function is best-effort

Comment (osherdp) [reply internal/imageservice/url_test.go:160]: yeah, just making it clearer

File: internal/bminventory/inventory_v2_handlers.go (hunk 2 above)
Comment (adriengentil) [reply internal/bminventory/inventory_v2_handlers.go:822]: `nil` should be returned in case of `byid` and `byapikey`, will fix it

Comment (adriengentil) [reply internal/bminventory/inventory_v2_handlers.go:822]: actually no, before we were calling `ParseExpirationFromURL`: https://github.com/openshift/assisted-service/pull/5523/files#diff-9f6fae8a28b132e15c426658b6b5d41adf8a20006ddeb271a22fea270c573610R85-R96

which returns the default value of `strfmt.DateTime` when no `image _token` is passed.

Comment (adriengentil) [reply internal/bminventory/inventory_v2_handlers.go:822]: We have tests here:
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18547
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18579
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory_test.go#L18601

Comment (osherdp) [reply internal/bminventory/inventory_v2_handlers.go:822]: so for all cases we're always returning an expiration date of `0001-01-01T00:00:00.000Z`?

Comment (adriengentil) [reply internal/bminventory/inventory_v2_handlers.go:822]: no, in case of `image_token` we don't expect `0001-01-01T00:00:00.000Z`

Comment (osherdp) [reply internal/bminventory/inventory_v2_handlers.go:822]: ack. thanks for the explanation!

**Review by osherdp — APPROVED** [review]

**Pull request merged**